### PR TITLE
python311Packages.dask-yarn: mark as unsupported on python 3.12

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

It seems to rely on `versioneer` stuff that is incompatible with python 3.12.
Note: There has been no releases since April 2021, so this package might be unmaintained (last commit from June 2022).

cc @illustris 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by illustris** [comment]
Versioneer supports python 3.12. The issue here is that the bundled versioneer.py in the repo is outdated. This can be fixed similar to this PR for skein: https://github.com/NixOS/nixpkgs/pull/310218/files#diff-86ca46f8580a8a9246614cee5e1fdba5b0a78aef74ab8ae283fe9fa6b588d2feR51

File: pkgs/development/python-modules/dask-yarn/default.nix [hunk 1]
```diff
@@ -7,6 +7,7 @@
 , fetchpatch
 , grpcio
 , pytestCheckHook
+, pythonAtLeast
```
Comment (illustris) [inline pkgs/development/python-modules/dask-yarn/default.nix:10]:
```suggestion
, versioneer
```

File: pkgs/development/python-modules/dask-yarn/default.nix [hunk 2]
```diff
@@ -16,7 +17,9 @@ buildPythonPackage rec {
   version = "0.9";
   format = "setuptools";
 
-  disabled = pythonOlder "3.7";
+  # Not compatible with python 3.12
+  # AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
+  disabled = pythonOlder "3.7" || pythonAtLeast "3.12";
```
Comment (illustris) [inline pkgs/development/python-modules/dask-yarn/default.nix:22]:
```suggestion
  disabled = pythonOlder "3.7";
  postPatch = "rm versioneer.py";
  build-system = [ versioneer ];
```

**Review by illustris — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/dask-yarn/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/dask-yarn/default.nix]: Indeed it works ! Thanks @illustris

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.dask-yarn</li>
    <li>python311Packages.dask-yarn.dist</li>
    <li>python312Packages.dask-yarn</li>
    <li>python312Packages.dask-yarn.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.dask-yarn</li>
    <li>python311Packages.dask-yarn.dist</li>
    <li>python312Packages.dask-yarn</li>
    <li>python312Packages.dask-yarn.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.dask-yarn</li>
    <li>python311Packages.dask-yarn.dist</li>
    <li>python312Packages.dask-yarn</li>
    <li>python312Packages.dask-yarn.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 310935` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.dask-yarn</li>
    <li>python311Packages.dask-yarn.dist</li>
    <li>python312Packages.dask-yarn</li>
    <li>python312Packages.dask-yarn.dist</li>
  </ul>
</details>

**Review by illustris — APPROVED** [review]
Thanks!

**Pull request merged**